idunn/utils/redis.py

```diff
@@ -25,3 +29,86 @@ def get_redis_pool(db):
         redis_url = "redis://" + redis_url
 
     return ConnectionPool.from_url(url=redis_url, socket_timeout=REDIS_TIMEOUT, db=db)
+
+
+class CacheNotAvailable(Exception):
+    pass
+
+
+class RedisWrapper:
+    _expire = None
+    _connection = None
+
+    @classmethod
+    def set_value(cls, key, json_result):
+        try:
+            cls._connection.set(key, json_result, ex=cls._expire)
+        except RedisError:
+            prometheus.exception("RedisError")
+            logging.exception("Got a RedisError")
+
+    @classmethod
+    def get_value(cls, key):
+        try:
+            value_stored = cls._connection.get(key)
+            return value_stored
+        except RedisError as exc:
+            prometheus.exception("RedisError")
+            logging.exception("Got a RedisError")
+            raise CacheNotAvailable from exc
+
+    @classmethod
+    def init_cache(cls, expire):
+        cls._expire = int(expire)  # seconds
+        redis_db = settings["WIKI_CACHE_REDIS_DB"]
+        try:
+            redis_pool = get_redis_pool(db=redis_db)
+        except RedisNotConfigured:
+            logger.warning("No Redis URL has been set for caching", exc_info=True)
+            cls._connection = DISABLED_STATE
+        else:
+            cls._connection = Redis(connection_pool=redis_pool)
+
+    @classmethod
+    def cache_it(cls, key, f, expire=settings["WIKI_CACHE_TIMEOUT"]):
```

Comment:
`expire` is not used if the redis connection is already initialized.
I think it should be passed to `set_value`